docs(custom-connector): step by step guide on how to create a custom connector based on the rest connector

mathias-vandaele opened the pull request:
## Description

<!-- Provide an overview of what to expect in the PR. -->
<!-- Relate or link the associated epic or task. -->
<!-- Add `@camunda/tech-writers` as reviewer to pull in a tech writer, or add your embedded tech writer. -->

Provides a step by step guide on how to create a custom connector based on the REST connector

## When should this change go live?

ASAP

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)
- [ ] This is already available but undocumented and should be released within a week. (add `available & undocumented` label)
- [ ] This is on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (create draft PR and/or add `hold` label)
- [ ] This is part of a scheduled **alpha or minor**. (add alpha or minor label)
- [ ] There is **no urgency** with this change (add `low prio` label)

## PR Checklist

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for **an upcoming minor release** and:
  - [ ] are in the `/docs` directory (version 8.8).
  - [ ] are in the `/versioned_docs/version-8.7/` directory (version 8.7).
- [ ] My changes are for an **already released minor** and are in a `/versioned_docs` directory.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

- [ ] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [ ] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [ ] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

akeller commented:
@mathias-vandaele, this is currently only changing the 8.8 docs. Do you also mean to change 8.7 and 8.6? I ask because you mention this is available but undocumented. What versions can you use it in?

mathias-vandaele commented:
> @mathias-vandaele, this is currently only changing the 8.8 docs. Do you also mean to change 8.7 and 8.6? I ask because you mention this is available but undocumented. What versions can you use it in?

Yes It should also be added for `8.6` and `8.7`, I would like to know if that is the right place to include it, I was messaging @mesellings about it 

I will add it to `8.6` and `8.7`, could you confirm this documentation has been added to the right section ?

mesellings commented:
@mathias-vandaele I'll take a look at this for you as discussed 👍

conceptualshark commented on docs/components/connectors/custom-built-connectors/create-connector-from-rest.md:
```diff
@@ -0,0 +1,114 @@
+
+8. **Add an Icon**
+
+   - Enhance your connector's visual appeal by adding an appropriate icon.
```
The image referenced here and on line 106 seemed to be missing - but I am not sure an image for each step is required. The UI is fairly descriptive, and the code screenshots could be provided in codeblocks.

conceptualshark reviewed (commented):
I fixed the build errors and formatting here to get things rolling, but had a few questions for organizing the content, and left a comment on the existing images.

The [Connector templates](https://docs.camunda.io/docs/next/components/connectors/custom-built-connectors/connector-templates/) section has similar guides for [generating](https://docs.camunda.io/docs/next/components/connectors/custom-built-connectors/connector-template-generator/) and [managing](https://docs.camunda.io/docs/next/components/connectors/manage-connector-templates/) Connectors. Is the first section of this guide (creating a new REST Connector template) something that could be folded into the existing content? It looks like these pages need a little updating, but that might already be on @mesellings radar.

This guide could then be simplified to providing only an example tutorial for creating and using the template, or else those examples could be nested into updates of the other content. 

Is there an issue or epic driving this change? Knowing the planning might help me understand how to organize the content, too.

mathias-vandaele commented:
Hey @mesellings 

I removed some pictures and added code blocks, 
I also added this doc into versioned doc, let me know if everything is alright. 👍

mesellings commented:
Thanks @mathias-vandaele I'll work on this now/today 👍

mesellings commented:
@mathias-vandaele I'll commit my changes directly in the branch, let me know if you haven't finished working on it?

mesellings reviewed (commented):
@mathias-vandaele I've completed a ton of restructuring, rewording and so on, but am happy to approve once we have finalised the caution (see comment) - great work to get all this content and page up and running 👍

mesellings commented on docs/components/connectors/custom-built-connectors/create-connector-from-rest.md:
```diff
@@ -0,0 +1,183 @@
+1. Open the new connector template in the template editor and customize it as required. For example, add or remove fields, adjust default values and input parameters, and update the description and other metadata.
+1. Save your changes to the new connector template, and use it as required in your processes.
+
+:::caution
+If a variable is used elsewhere in your element template, the variable using other variables is declared after. Otherwise, they will be considered `null`.
+:::
```
@mathias-vandaele I'm not sure how to reword this as I'm unclear what the sentence is meant to convey? Could you please reword or add clarity and I can amend? Thanks!

mathias-vandaele replied:
I had an hard time explaining this.

Basically when you construct a new element template base on REST, you need to put the field that set variables BEFORE the fields that are using those variables, 

example:

This works

```
{
      "id": "swid",
      "label": "id",
      "description": "Index of the resource",
      "feel": "optional",
      "group": "swapi",
      "binding": {
        "name": "index",
        "type": "zeebe:input"
      },
      "type": "String"
    },
    {
      "id": "swresource",
      "label": "Type",
      "description": "Choose the resource type",
      "value": "Planets",
      "group": "swapi",
      "binding": {
        "name": "resource",
        "type": "zeebe:input"
      },
      "type": "Dropdown",
      "choices": [...]
    },
    {
      "id": "url",
      "label": "URL",
      "optional": false,
      "constraints": {
        "notEmpty": true,
        "pattern": {
          "value": "^(=|(http://|https://|secrets|\\{\\{).*$)",
          "message": "Must be a http(s) URL"
        }
      },
      "group": "endpoint",
      "binding": {
        "name": "url",
        "type": "zeebe:input"
      },
      "type": "Hidden",
      "value": "=\"https://swapi.dev/api/\" + resource + \"/\" + index"
    }
```

This does not


```
    {
      "id": "url",
      "label": "URL",
      "optional": false,
      "constraints": {
        "notEmpty": true,
        "pattern": {
          "value": "^(=|(http://|https://|secrets|\\{\\{).*$)",
          "message": "Must be a http(s) URL"
        }
      },
      "group": "endpoint",
      "binding": {
        "name": "url",
        "type": "zeebe:input"
      },
      "type": "Hidden",
      "value": "=\"https://swapi.dev/api/\" + resource + \"/\" + index"
    },
     {
      "id": "swid",
      "label": "id",
      "description": "Index of the resource",
      "feel": "optional",
      "group": "swapi",
      "binding": {
        "name": "index",
        "type": "zeebe:input"
      },
      "type": "String"
    },
    {
      "id": "swresource",
      "label": "Type",
      "description": "Choose the resource type",
      "value": "Planets",
      "group": "swapi",
      "binding": {
        "name": "resource",
        "type": "zeebe:input"
      },
      "type": "Dropdown",
      "choices": [...]
    }
```


`url` needs to be placed AFTER because it is using variables defined by `swid` and `swresource`

mesellings replied:
Ah, I see - thanks - I'll adjust the note 👍

mesellings commented:
@mathias-vandaele I've adjusted that caution in the latest commit - please re-review and approve if you're happy for this to go out 👍

mesellings reviewed (approved):
Lgtm! 🚀
